docs/DeveloperGuide.md

```diff
@@ -1,34 +1,1260 @@
+ `Task`s, and a few helper methods.
+
+This means that multiple operations such as addition and deletion can be done on a `Task`, without affecting
+the contents of other `Task` in the `TaskList`.
+<br>
+
+Given below is an example usage scenario and how the add task function behaves.
+
+1. The user launches the application for the first time. CliUserInterface#executeCommand is called when the user adds a `Task` into the `TaskList`.
+
+2. The user types `add /tTask /by19-10-2020 1900`. The `add` command is passed through `InputParser#getCommandFromInput`, which then creates a constructor for `AddCommandTimetableMode`.
+
+3. `AddCommandTimetableMode#execute()` is called, which then calls `InputParser#parseTaskTitle`, which first extracts the `title` from the user's input.
+
+4. `InputParser#parseDeadline` is then called, which returns the `deadline` to `AddCommandTimetableMode#execute()`.
+
+5. `TaskList#addTask` is then called and a new `Task`, with `title` and `deadline`, is initialised.
+
+6. To signal that the user has successfully added a task, a message is printed with `CliMessages#printAddedTaskMessage`.
+
+The UML sequence diagram below shows how the add task command works.
+![Sequence Diagram for Add Task Command](diagrams/class/jpeg/add_task.jpg)
+
+<hr>
+TaskList also allows the deletion of tasks by the user.
+
+The figure below shows how the delete task command works:
+![Sequence Diagram for Delete Task Command](diagrams/class/jpeg/SequenceDiagram_RemoveTimetable.jpg)
+
+Here are the general steps that the command goes through when the user inputs "delete 1":
+1. The `CliUserInterface` receives the "delete 1" input by the user and passes it to the `InputParser` class.
+2. `InputParser` parses the input to determine the type of command and the index of the task that is required to delete.
+The Parser then constructs a `RemoveCommandTimetableMode` with constructor as shown below.
+```java
+public RemoveCommandTimetableMode(int indexToRemove, AppState uiMode) {
+    this.setAppState(uiMode);
+    this.indexToRemove = indexToRemove;
+}
+```
+3. Method **execute()** then calls the `TaskList` stored in `AppState` to update the deletion of the task.
+It also constructs `CliMessages` to display messages to the user.
+4. If the deletion is successful, `CliMessages` displays the message to the user.
+
+##### 4.2.1.2. Design Considerations
+
+###### Aspect: How to store tasks in `TaskList`
+- **Alternative 1 (current choice):** Store as an `ArrayList` of tasks
+    - Pros: It is easier to implement because the code base are list based.
+    - Cons: It is unoptimized in terms of complexity, which requires more work for scaling of the application.
+- **Alternative 2:** Store as a Hash Table with the key as the index and value as `Task`
+    - Pros: It has a better time complexity and reduce the work in scaling stage since this data structure is more optimized (O(1) can be achieved).
```

Comment:
Are you referring to space or time complexity? You could use an array of a fixed size, lets say 10000 and store it by index. Of course, the number of tasks you can create is limited but since its going to be small. You might want to explore this?